### PR TITLE
fix: Helm v3 CRD are not deployed

### PR DESCRIPTION
**Pull request by mayzhang2000**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: util/helm/cmd.go [hunk 1]
```diff
@@ -221,6 +221,9 @@ func (c *Cmd) template(chartPath string, opts *TemplateOpts) (string, error) {
 	for _, v := range opts.APIVersions {
 		args = append(args, "--api-versions", v)
 	}
+	if c.HelmVer.additionalTempalteArgs != nil {
```
Comment (jannfis) [inline util/helm/cmd.go:224]: I guess there is a typo in the field name, `additionalTempalteArgs` should really be `additionalTemplateArgs`

File: util/helm/helmver.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ var (
 		pullCommand:             "pull",
 		initSupported:           false,
 		getPostTemplateCallback: cleanupChartLockFile,
+		additionalTempalteArgs:  []string{"--include-crds"},
```
Comment (jannfis) [inline util/helm/helmver.go:32]: Typo :)

File: util/helm/helmver.go [hunk 3]
```diff
@@ -82,4 +83,5 @@ type HelmVer struct {
 	pullCommand             string
 	kubeVersionSupported    bool
 	getPostTemplateCallback func(chartPath string) (func(), error)
+	additionalTempalteArgs  []string
```
Comment (jannfis) [inline util/helm/helmver.go:86]: Typo :)

**Review by jannfis — CHANGES_REQUESTED** [review]
Looks good so far, just minor comments about a typo

File: util/helm/cmd.go (hunk 1 above)
Comment (mayzhang2000) [reply util/helm/cmd.go]: Thank you! Fixed. :)

File: util/helm/helmver.go (hunk 2 above)
Comment (mayzhang2000) [reply util/helm/helmver.go]: Fixed.

File: util/helm/helmver.go (hunk 3 above)
Comment (mayzhang2000) [reply util/helm/helmver.go]: Fixed

**Comment by jannfis** [comment]
Thanks @mayzhang2000 - I just got one additional comment:

Maybe it would make sense in the corresponding E2E test to check that the CRD has actually been synced correctly, using maybe `ResourceSyncStatusIs()` expectation.

**Review by jannfis — APPROVED** [review]
Awesome, mayzhang2000! LGTM now!

**Review by alexmt — APPROVED** [review]
Thank you for review @jannfis !

**Comment by mayzhang2000** [comment]
closes #3336

**Pull request merged**